src/pkg/bundle/deploy.go

```diff
@@ -443,9 +446,19 @@ func addOverrideValue(overrides map[string]map[string]*values.Options, component
 			templatedVariable := fmt.Sprintf("%v", v)
 			value = setTemplatedVariables(templatedVariable, pkgVars)
 		}
-		// handle default case of simple values like strings and numbers
-		helmVal := fmt.Sprintf("%s=%v", valuePath, value)
-		overrides[component][chart].Values = append(overrides[component][chart].Values, helmVal)
+
+		// Check for files else handle default case of simple values like strings and numbers
+		if valueType == "file" {
```

Comment:
I think you could pull this whole `if` block up above the switch statement and wrap it in a function. This would remove the need for the `else` clause

Reply:
i like that idea. Sorry if i'm misunderstanding the ask, but sounds like your suggesting wrapping the entire conditional in a function. wrapping everything though, even the valuteType check, allows for some misdirection in that after adding the file to the `.FileValues` it will continue on to the switch, fall through to the default case and be added to `.Values` as well. It looks like things still work out because Helm evaluates `FileValues` after `Values` so the key gets set back to the file. But maybe to avoid a potential issue and confusion, we should leave the valueType check outside of the file handler function and return if it was a file before hitting the switch.

Reply:
You nailed it, I should've used more words. You can leave the if statement as is, process the files inside it, and return